### PR TITLE
DA-3168: Create Backend/ Alembic NPH Structure

### PR DESCRIPTION
**Pull request by jmtrujillo75**
## Description of changes/additions
1. Added `./tools/generate_schema_rex.sh` script to generate schema revisions for `rex` database
2. Added `./tools/generate_schema_nph.sh` script to generate schema revisions for `nph` database 
3. Added `./tools/upgrade_database_nph.sh` script to upgrade schema for schema revisions for `nph` database
4. Added `./tools/upgrade_database_rex.sh` script to upgrade schema for schema revisions for `rex` database
5. Updated `./tools/setup_local_database.sh` script to upgrade the schema for both `rex` and `nph` databases after upgrading the `rdr` & `metrics` databases 

### **Generate First Test Schema for `Rex` database**

```zsh
$ ./tools/generate_schema_rex.sh "initial"
13:39:04 INFO  [alembic.env] Migrating database rex
13:39:04 INFO  [alembic.runtime.migration] Context impl MySQLImpl.
13:39:04 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
13:39:04 INFO  [alembic.autogenerate.compare] Detected added table 'test'
  Generating /Users/trujiljm/VUMC_Projects/raw-data-
  repository/rdr_service/alembic/rex/versions/abbfda97030b_initial.py ...  done
  Running post write hook "custom_sql_newlines" ...
  done
```

### **Apply migrations to the `rex` database**

```zsh
$ ./tools/upgrade_database_rex.sh
13:39:13 INFO  [alembic.env] Migrating database rex
13:39:13 INFO  [alembic.runtime.migration] Context impl MySQLImpl.
13:39:13 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
13:39:13 INFO  [alembic.runtime.migration] Running upgrade  -> abbfda97030b, initial
```

### **Verification**

```zsh
$ mysql -h 127.0.0.1 -P 3306 -u root -p
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 188
Server version: 5.7.39 Homebrew

Copyright (c) 2000, 2022, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> use rex;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> show tables;
+-----------------+
| Tables_in_rex   |
+-----------------+
| alembic_version |
| test            |
+-----------------+
2 rows in set (0.00 sec)
```

### **Generate First Test Schema for `nph` database**

```zsh
./tools/generate_schema_nph.sh "initial"
13:49:58 INFO  [alembic.env] Migrating database nph
13:49:58 INFO  [alembic.runtime.migration] Context impl MySQLImpl.
13:49:58 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
13:49:58 INFO  [alembic.autogenerate.compare] Detected added table 'test'
  Generating /Users/trujiljm/VUMC_Projects/raw-data-
  repository/rdr_service/alembic/nph/versions/6735911888a8_initial.py ...  done
  Running post write hook "custom_sql_newlines" ...
  done
```

### **Apply migrations to the `nph` database**

```zsh
./tools/upgrade_database_nph.sh
13:50:08 INFO  [alembic.env] Migrating database nph
13:50:08 INFO  [alembic.runtime.migration] Context impl MySQLImpl.
13:50:08 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
13:50:08 INFO  [alembic.runtime.migration] Running upgrade  -> 6735911888a8, initial
```

## **Verification**

```zsh
$ mysql -h 127.0.0.1 -P 3306 -u root -p
Enter password:
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 188
Server version: 5.7.39 Homebrew

Copyright (c) 2000, 2022, Oracle and/or its affiliates.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> use nph;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed
mysql> show tables;
+-----------------+
| Tables_in_nph   |
+-----------------+
| alembic_version |
| test            |
+-----------------+
2 rows in set (0.00 sec)
```

### **Generate Schema Test for `rdr` & `metrics` database**

```zsh
$ ./tools/generate_schema.sh "added isGhostIdDuplicate field"
RDR Password rdr!pwd
11:36:55 INFO  [alembic.env] Migrating database rdr
11:36:55 INFO  [alembic.runtime.migration] Context impl MySQLImpl.
11:36:55 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
11:36:55 INFO  [alembic.autogenerate.compare] Detected added column 'participant.is_ghost_id_duplicate'
11:36:55 INFO  [alembic.autogenerate.compare] Detected added column 'participant_history.is_ghost_id_duplicate'
/Users/trujiljm/VUMC_Projects/raw-data-repository/venv39/lib/python3.9/site-packages/alembic/autogenerate/compare.py:969: UserWarning: Computed default on participant_summary.health_data_stream_sharing_status_v_3_1 cannot be modified
  util.warn("Computed default on %s.%s cannot be modified" % (tname, cname))
/Users/trujiljm/VUMC_Projects/raw-data-repository/venv39/lib/python3.9/site-packages/alembic/autogenerate/compare.py:969: UserWarning: Computed default on participant_summary.health_data_stream_sharing_status_v_3_1_time cannot be modified
  util.warn("Computed default on %s.%s cannot be modified" % (tname, cname))
/Users/trujiljm/VUMC_Projects/raw-data-repository/venv39/lib/python3.9/site-packages/alembic/autogenerate/compare.py:969: UserWarning: Computed default on participant_summary.was_ehr_data_available cannot be modified
  util.warn("Computed default on %s.%s cannot be modified" % (tname, cname))
11:36:55 INFO  [alembic.env] Migrating database metrics
11:36:55 INFO  [alembic.runtime.migration] Context impl MySQLImpl.
11:36:55 INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
11:36:55 INFO  [alembic.env] skipping not allowed aggregate_metrics_ibfk_1
  Generating /Users/trujiljm/VUMC_Projects/raw-data-repository/rdr_service/alembic/versions/ef3367cd2c3b_added_isghostidduplicate_field.py ...  done
  Running post write hook "custom_sql_newlines" ...
  done
```
Just from above results we can infer that `rdr` & `metrics` related scripts `./tools/generate_schema.sh` & `./tools/upgrade_database.sh` remain unchanged and should work as expected.

## Tests
N/A

### CODE REVIEW

**Review by j-kanuch — APPROVED** [review]

File: rdr_service/model/rex.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+from rdr_service.model.base import RexBase
+
+
+class TestModel(RexBase):
+
+    __tablename__ = "test"
+
+    id = Column("id", Integer, primary_key=True)
+    name = Column("name", String(256), nullable=False)
+    place = Column("place", String(256), nullable=True)
```
Comment (kenny-skaggs) [inline rdr_service/model/rex.py]: I appreciate seeing that we're able to generate migrations for the new databases. But do we need to have these test classes and their migration files be merged? It seems like we're setting ourselves up to need to create more migrations to remove the tables later, and we'll always have the test table in the migration files.

Comment (jmtrujillo75) [reply rdr_service/model/rex.py]: I mean as soon as legitimate model for `Nph` & `Rex` databases is added we can remove the test model. @j-kanuch should we remove the `TestModel` model from the commit history now itself.

Comment (jmtrujillo75) [reply rdr_service/model/rex.py]: If the notes in the above PR description should suffice for now, I will remove the `Test` model definitions and related migration files. Not sure if `Github` will still display the empty `alembic/nph/versions/` & `alembic/rex/versions/` folders in the repo.

Comment (kenny-skaggs) [reply rdr_service/model/rex.py]: I'm still leaning towards removing the Test class and migrations. I don't think git would keep the alembic/nph and /rex folders, but we can recreate them when we set up the initial migrations (if alembic doesn't create them on its own).

Comment (jmtrujillo75) [reply rdr_service/model/rex.py]: I removed the `test` table migrations

**Review by kenny-skaggs — APPROVED** [review]

**Pull request merged**